src/WingetCreateCLI/Commands/BaseCommand.cs

```diff
@@ -256,6 +257,24 @@ protected static bool ValidateManifestsInTempDir(Manifests manifests)
             return result;
         }
 
+        /// <summary>
+        /// Displays the appropriate warning messages for installers with detected architecture mismatches.
+        /// </summary>
+        protected static void DisplayMismatchedArchitectures()
+        {
+            if (PackageParser.ArchMismatches.Any())
+            {
```

Comment:
I don't like the idea of making PackageParser stateful. Is there any reason you can't just have DisplayMismatchedArchitectures take the mismatches as a parameter?

Reply:
Changed displaymismatchedarchitectures to take in a parameter. Created an out variable to pass along the detected architectures.